packages/components/src/internal/components/editable/models.ts

```diff
@@ -344,6 +345,12 @@ export class EditorModel
                     let val;
                     if (values.size === 1) val = values.first()?.raw;
                     row = row.set(col.name, val);
+
+                    // Issue 39517: include display value in the row data for sample lookup columns
```

Comment:
It's always a little peculiar to me when the client tells the server about something it already should know / be able to figure out. I realize the context is lacking on the server-side in this case but that feels like the crux of the problem where this could be solved more broadly.

Reply:
Yes, I am not thrilled about this solution either. This was definitely a compromise solution since it would take a lot more code on the server side to get the specific context it needs to determine that this is a sample lookup and to pull in the sample name via the rowId lookup. I'm happy to discuss alternate solutions here if you have ideas.